internal/compressor/flate.go

```diff
@@ -0,0 +1,65 @@
+package compressor
+
+import (
+	"bytes"
+	"compress/flate"
+	"fmt"
+	"github.com/lf-edge/ekuiper/internal/conf"
+	"io"
+)
+
+func newFlateCompressor() (*flateCompressor, error) {
+	flateWriter, err := flate.NewWriter(nil, flate.DefaultCompression)
+	if err != nil {
+		return nil, err
+	}
+	return &flateCompressor{
+		writer: flateWriter,
+	}, nil
+}
+
+type flateCompressor struct {
+	writer *flate.Writer
+	buffer bytes.Buffer
+}
+
+func (g *flateCompressor) Compress(data []byte) ([]byte, error) {
+	g.buffer.Reset()
+	g.writer.Reset(&g.buffer)
+	_, err := g.writer.Write(data)
+	if err != nil {
+		return nil, err
+	}
+	err = g.writer.Close()
+	if err != nil {
+		return nil, err
+	}
+	return g.buffer.Bytes(), nil
+}
+
+func newFlateDecompressor() (*flateDecompressor, error) {
+	return &flateDecompressor{}, nil
+}
+
+type flateDecompressor struct {
+	reader io.ReadCloser
```

Comment:
In the https://pkg.go.dev/compress/flate reset example, it is possible to init a reader with nil data, I suggest to follow the code style there. In zlib, NewReaer(nil) has an error, that's why I coded like this.

Reply:
got it , and why is this test failed ? 

```
mock sink receive  map[email:Dddd id:1 name:test]
--- FAIL: TestFormat_Apply (5.51s)
    sink_node_test.go:315: 1 	result mismatch:
        
        exp=[0a047465737410011a0444646464]
        
        got=[]
        
The test bucket size is 6.
```

Reply:
> In the https://pkg.go.dev/compress/flate reset example, it is possible to init a reader with nil data, I suggest to follow the code style there. In zlib, NewReaer(nil) has an error, that's why I coded like this.

code modified , and the size is kuiper 7.6M and kuiperd 28.6M  , same as before

Reply:
> got it , and why is this test failed ?
> 
> ```
> mock sink receive  map[email:Dddd id:1 name:test]
> --- FAIL: TestFormat_Apply (5.51s)
>     sink_node_test.go:315: 1 	result mismatch:
>         
>         exp=[0a047465737410011a0444646464]
>         
>         got=[]
>         
> The test bucket size is 6.
> ```

Ah, it is a problem of the test case which is synced by time.sleep now.. Not your problem